app/src/main/java/com/github/se/eventradar/model/EventsOverviewViewModel.kt

```diff
@@ -0,0 +1,96 @@
+
+    fun getEvents() {
+        eventRef
+            .get()
+            .addOnSuccessListener { result ->
+                val events =
+                    result.documents.map { document ->
+                        Event(
+                            eventName = document.data?.get("name") as String,
+                            eventPhoto = document.data?.get("photo_url") as String,
+                            start = getLocalDateTime(document.data?.get("start") as String),
+                            end = getLocalDateTime(document.data?.get("end") as String),
+                            location =
+                                getLocation(
+                                    document.data?.get("location_name") as String,
+                                    document.data?.get("location_lat") as Double,
+                                    document.data?.get("location_lng") as Double,
+                                ),
+                            description = document.data?.get("description") as String,
+                            ticket = getEventTicket(
+                                document.data?.get("ticket_name") as String,
+                                document.data?.get("ticket_price") as Double,
+                                document.data?.get("ticket_quantity") as Int,
+                            ),
+                            contact = document.data?.get("contact_email") as String,
+                            organiserList = getSet(document.data?.get("organisers_list")),
+                            attendeeList = getSet(document.data?.get("attendees_list")),
+                            category = getEventCategory(document.data?.get("category") as String),
+                            fireBaseID = document.id
+                        )
+                    }
+                _uiState.value =
+                    _uiState.value.copy(
+                        eventList = EventList(events, events, _uiState.value.eventList.getEvent))
+            }
+            .addOnFailureListener { exception ->
+                Log.d("EventsOverviewViewModel", "Error getting documents: ", exception)
+            }
+    }
+    fun onTaskClicked(id: String) {
+        // Temporarily commented out as we transition from handling to-dos to handling events.
+
+        /*
+        val to_Do = _uiState.value.toDoList.getAllTask.find { it.id == id }
+        _uiState.value =
+            _uiState.value.copy(
+                toDoList =
+                    ToDoList(
+                        _uiState.value.toDoList.getAllTask,
+                        _uiState.value.toDoList.getFilteredTask,
+                        to_Do))
+           */
+    }
+
+    companion object {
+        fun getLocalDateTime(dateTime: String): LocalDateTime {
+            val formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss")
+            return LocalDateTime.parse(dateTime, formatter)
+        }
+
+        fun getSet(data: Any?): Set<String> {
```

Comment:
What does this function do? The name is not specific at all.

Reply:
It gets from the firebase a set of strings - have renamed for clarity